### PR TITLE
Fix more #121208 fallout

### PR DESCRIPTION
**Pull request by nnethercote**
#121208 converted lots of delayed bugs to bugs. Unsurprisingly, there were a few invalid conversion found via fuzzing.

r? @lcnr

### CODE REVIEW

**Comment by nnethercote** [comment]
Note: the test for #121457 hit two different `span_bug`s in `refine.rs`.

File: compiler/rustc_hir_typeck/src/expr.rs [hunk 1]
```diff
@@ -76,9 +76,16 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
...
-            if let Some(_) = self.typeck_results.borrow().adjustments().get(expr.hir_id) {
-                self.dcx()
-                    .span_bug(expr.span, "expression with never type wound up being adjusted");
+            if let Some(adjustments) = self.typeck_results.borrow().adjustments().get(expr.hir_id) {
+                let reported = self.dcx().span_delayed_bug(
+                    expr.span,
+                    "expression with never type wound up being adjusted",
+                );
+                return if let [Adjustment { kind: Adjust::NeverToAny, target }] = &adjustments[..] {
+                    target.to_owned()
```
Comment (lcnr) [inline compiler/rustc_hir_typeck/src/expr.rs:85]: is the error message worse without that `if let`?

if not please remove it

**Review by lcnr — COMMENTED** [review]
r=me regardless of what you do with the nit

File: compiler/rustc_hir_typeck/src/expr.rs (hunk 1 above)
Comment (nnethercote) [reply compiler/rustc_hir_typeck/src/expr.rs]: Just doing `Ty::new_error` has no effect on test outputs, so I did this.

**Pull request merged**